types/ui/searchFilter.d.ts

```diff
@@ -1 +1,75 @@
-declare class SearchFilter {}
+/**
+ * A controller class for managing a text input widget that filters the contents of some other UI element
+ * @see {@link Application}
+ *
+ */
+declare class SearchFilter {
+  /**
+   * @param inputSelector   - The CSS selector used to target the text input element.
+   * @param contentSelector - The CSS selector used to target the content container for these tabs.
+   * @param initial         - The initial value of the search query.
+   *                          (default: `''`)
+   * @param callback        - A callback function which executes when the filter changes.
+   * @param delay           - The number of milliseconds to wait for text input before processing.
+   *                          (default: `100`)
+   */
+  constructor({
+    inputSelector,
+    contentSelector,
+    initial,
+    callback,
+    delay
+  }: {
+    inputSelector: string;
+    contentSelector: string;
+    initial?: SearchFilter['query'];
+    callback: SearchFilter['callback'];
+    delay?: number;
+  });
+
+  /**
+   * The value of the current query string
+   */
+  query: string;
+
+  /**
+   * A callback function to trigger when the tab is changed
+   */
+  callback: (event: KeyboardEvent, query: string, content: string) => void | null;
```

Comment:
`null` is a valid value for `void`, so `void | null === void`. You could consider making the return type `unknown` because it doesn't matter at all.

Reply:
Actually that was me mistyping. This started out as `Function | null` and I only expanded the `Function` when I figured out what it was getting passed. But now that I look at `bind()` again, `callback` should not be allowed to be `null`.